viite-backend/viite-main/src/main/scala/fi/vaylavirasto/viite/dynamicnetwork/LinkNetworkUpdater.scala

```diff
@@ -0,0 +1,746 @@
+   */
+  private def convertToAValidSplitChange(change: LinkNetworkChange): Option[LinkNetworkSplitChange] = {
+
+    // Validate the change for a split change; get out with a throw if the change is not a valid split change
+
+    if(!change.changeType.equals("split")) {
+      throw ViiteException(s"LinkNetworkChange: Shall not try to convert a change of type ${change.changeType} to a LinkNetworkSplitChange. " +
+        s"Check the parameter of the calling function.")
+    }
+
+    logger.debug("size considerations")
+    if ( change.newLinks.size <= 2
+      || change.replaceInfos.size <= 2
+      || change.newLinks.size != change.replaceInfos.size
+    ) {
+      throw ViiteException(s"LinkNetworkChange: Invalid SplitChange. A split must have at least " +
+        s"two new links, and their corresponding replace infos (who may be further splitted to smaller chunks). " +
+        s"There are ${change.newLinks.size} new links, and ${change.replaceInfos.size} replace infos " +
+        s"when going to split ${change.oldLink.linkId}.")
+    }
+
+    // Ok, now we know we have a proper amount of components
+    val oldLink = change.oldLink
+    val newLinks = change.newLinks
+    val splitInfos = change.replaceInfos
+
+    logger.debug("data integrity: link ids")
+    if (!splitInfos.forall(si => oldLink.linkId == si.oldLinkId)) {
+      throw ViiteException(s"LinkNetworkChange: Invalid SplitChange. The old link ${oldLink.linkId} " +
+        s"must be part of all of the replace infos. ")
+    }
+    newLinks.foreach(nl =>
+      if(!splitInfos.exists(si => nl.linkId==si.newLinkId)) {
+        throw ViiteException(s"LinkNetworkChange: Invalid SplitChange. Correspondence between " +
+          s"the new link ${nl.linkId}, and a replace info link is not found. ")
+      }
+    )
+
+    // -- data integrity - lengths --
+    val oldLengthFromOldLink    = oldLink.linkLength
+    val oldLengthFromSplitInfos = splitInfos.foldLeft(0.0)((cumulLength,splitInfo) => cumulLength + splitInfo.oldToMValue-splitInfo.oldFromMValue)
+    val newLengthFromNewLinks   = newLinks  .foldLeft(0.0)((cumulLength,splitInfo) => cumulLength + splitInfo.linkLength)
+    val newLengthFromSplitInfos = splitInfos.foldLeft(0.0)((cumulLength,splitInfo) => cumulLength + splitInfo.newToMValue-splitInfo.newFromMValue)
+
+    logger.debug(s"data integrity: lengths must match ") // must match sufficiently. Allowed difference: ${GeometryUtils.DefaultEpsilon} m ")
+    if (oldLengthFromOldLink  != oldLengthFromSplitInfos) {  // old link lengths must always match
+      throw ViiteException(s"LinkNetworkChange: Invalid SplitChange. Old link lengths do not match when splitting link $oldLink." +
+        s"Check that lengths (oldToMValue-oldFromMValue) in the replaceInfos (now $oldLengthFromSplitInfos) " +
+        s"sum up to that of the old link length ($oldLink.linkLength)")
+    }
+    if (newLengthFromNewLinks != newLengthFromSplitInfos) { // new link lengths must always match
```

Comment:
Uuden linkin pituus tarkistuksissa, ReplaceInfoista yhteenlaskettu uuden linkin pituus pitää koostua kaikista LinkNetworkChange muutossanomista. 

Tämä siitä syystä että uusien linkkien Tiekamusanomista johdetut ReplaceInfot voivat olla hajautuneina useisiin eri LinkNetworkChange muutossanomiin. 

Jolloin yhden LinkNetworkChange muutossanoman sisällä olevat ReplaceInfot eivät kerro uuden linkin koko pituutta vaan ainoastaan sen osuuden pituuden jota siinä kyseisessä muutossanomassa käsitellään.

Reply:
Bump

Reply:
Hmm. Eli tää pitäis nostaa jonnekin ylemmälle tasolle.
Tässä ratkaisu kuitenkin lienee, että tästä pois.